From 13f897d0324ff51a2de472ec8b745eb704ed4fb5 Mon Sep 17 00:00:00 2001
From: huntxu <mhuntxu@gmail.com>
Date: Thu, 7 May 2015 02:19:45 +0800
Subject: [PATCH 2/2] Add external-fixed-ip to router-gateway-set

a parameter is added to the API.
this patch adds a parameter that is collesponding to the parameter.

Signed-off-by: huntxu <mhuntxu@gmail.com>
---
 neutronclient/neutron/v2_0/router.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/neutronclient/neutron/v2_0/router.py b/neutronclient/neutron/v2_0/router.py
index 369703e..e030cd2 100644
--- a/neutronclient/neutron/v2_0/router.py
+++ b/neutronclient/neutron/v2_0/router.py
@@ -187,6 +187,9 @@ class SetGatewayRouter(neutronV20.NeutronCommand):
         parser.add_argument(
             '--disable-snat', action='store_true',
             help=_('Disable source NAT on the router gateway.'))
+        parser.add_argument(
+            '--external-fixed-ip',
+            help=_('Fixed external IP address of the router.'))
         return parser
 
     def run(self, parsed_args):
@@ -200,6 +203,9 @@ class SetGatewayRouter(neutronV20.NeutronCommand):
         router_dict = {'network_id': _ext_net_id}
         if parsed_args.disable_snat:
             router_dict['enable_snat'] = False
+        if parsed_args.external_fixed_ip:
+            router_dict['external_fixed_ips'] = [
+                {'ip_address': parsed_args.external_fixed_ip}]
         neutron_client.add_gateway_router(_router_id, router_dict)
         print(_('Set gateway for router %s') % parsed_args.router_id,
               file=self.app.stdout)
-- 
2.4.0

